### PR TITLE
Add back depends_on directives needed to bootstrap on Python 3.6

### PR DESCRIPTION
**Pull request by alalazo**
Reverts spack/spack#34137

@adamjstewart Trying to check if this broke bootstrapping when merged to develop

### CODE REVIEW

**Comment by adamjstewart** [comment]
@tgamblin when is that merge queue coming? 😅

**Comment by alalazo** [comment]
Arrgh, I need to push also some silly change to core to run what I want to check :laughing:

**Comment by alalazo** [comment]
@adamjstewart Ok, this is working as expected. Wondering what is the best way to check that bootstrapping on Python 3.6 continues working when there's a change in a Python related package. :thinking:

**Comment by alalazo** [comment]
Also, do you want me to bring back in the `PythonExtension` modification? I think it's just dropping the `depends_on` that broke bootstrapping.

File: lib/spack/spack/build_systems/python.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def import_modules(self):
         in the source tarball directory. If the module names are incorrectly
         detected, this property can be overridden by the package.
 
-        Returns:
+        Return:
```
Comment (adamjstewart) [inline lib/spack/spack/build_systems/python.py]: Returns is correct

**Comment by adamjstewart** [comment]
Yes, let's keep PythonExtension.

For testing, can we test bootstrapping for both the latest and oldest versions of Python?

File: lib/spack/spack/build_systems/python.py (hunk 1 above)
Comment (alalazo) [reply lib/spack/spack/build_systems/python.py]:
```suggestion
        Returns:
```

Comment (alalazo) [reply lib/spack/spack/build_systems/python.py]: Yes, I just needed something to trigger unit tests

**Comment by alalazo** [comment]
@adamjstewart If this is fine with you I'll merge this hotfix for develop.

**Pull request merged**